servers/src/main/java/tachyon/master/file/FileSystemMaster.java

```diff
@@ -1094,4 +1120,30 @@ public void reportLostFile(long fileId) throws FileDoesNotExistException {
       return mDependencyMap.getPriorityDependencyList();
     }
   }
+
+  /**
+   * MasterInodeTTL periodic check.
+   */
+  public final class MasterInodeTTLCheckExecutor implements HeartbeatExecutor {
+    @Override
+    public void heartbeat()  {
+      synchronized (mInodeTree) {
+        List<Inode> inodes = mInodeTree.getInodeChildrenRecursive(mInodeTree.getRoot());
```

Comment:
that sounds a good idea. let me work on the bucketized ttl. deletion can be delayed until the bucket is expired. maybe for accessing file, we can also check the ttl to make sure we didn't return thing unexpected.